File: 02_Searching_Algorithms/Linear_Search_Algorithm/linear_search.js
/**
 * The function "search" takes an array, the length of the array, and a value to search for, and
 * returns the index of the value in the array if found, otherwise it returns -1.
 * @param arr - The arr parameter is an array of elements that we want to search through.
 * @param n - The parameter "n" represents the length of the array "arr".
 * @param x - The value we are searching for in the array.
 * @returns the index of the element 'x' in the array 'arr'. If the element is not found, it returns
 * -1.
 */
function search(arr, n, x) 
{ 
	for (let i = 0; i < n; i++) 
		if (arr[i] == x) 
			return i; 
	return -1; 
} 


/* The code is performing a linear search in an array to find a specific element. */
	let arr = [ 2, 3, 4, 10, 40 ]; 
	let x = 10; 
	let n = arr.length; 

	// Function call 
	let result = search(arr, n, x); 
	(result == -1) 
		? console.log("Element is not present in array") 
		: console.log("Element is present at index " + result); 

File: 02_Searching_Algorithms/Linear_Search_Algorithm/linear_search.py
"""
    The search function takes an array, its size, and a target value as input, and returns the index of
    the target value in the array if found, otherwise it returns -1.
    
    :param arr: The arr parameter is a list or array of elements in which we want to search for a
    specific element
    :param N: The parameter N represents the size of the array. It indicates the number of elements in
    the array that we want to search through
    :param x: The value we are searching for in the array
    :return: the index of the element 'x' in the array 'arr'. If 'x' is not found in the array, it
    returns -1.
"""

def search(arr, N, x): 

	for i in range(0, N): 
		if (arr[i] == x): 
			return i 
	return -1


# The code block `if __name__ == "__main__":` is used to check if the current script is being run
# directly or being imported as a module.
if __name__ == "__main__": 
	arr = [2, 3, 4, 10, 40] 
	x = 10
	N = len(arr) 

	# Function call 
	result = search(arr, N, x) 
	if(result == -1): 
		print("Element is not present in array") 
	else: 
		print("Element is present at index", result) 
